CONTRIBUTING.md

```diff
@@ -41,90 +41,95 @@ Unit tests are nice for atomic components, but since jdbi is a complex ecosystem
 
 Since our tests essentially describe and verify jdbi's behavior, changing their specifics where it isn't inherently necessary is considered a red flag.
 
+The use of mocks and mocking framework is generally discouraged. There are a number of existing tests that use Mockito and they are a pain to maintain.
+
 ## Pull requests
 
-Expect your pull request to be scrutinized in even the tiniest details (down to grammar in javadoc), and to need to address many remarks even for small changes. We strive for a healthy balance between subjective perfection and practical considerations, but we are firmly against doing a quick and sloppy job that will require a lot of follow-up work later.
+We strive for a healthy balance between subjective perfection and practical considerations, but we are firmly against doing a quick and sloppy job that will require a lot of follow-up work later.
 
 Due to the volume of feedback in a typical PR, we may push changes directly to your PR branch if we are able to, in order to save time and frustration for everyone.
 
-# Setup
-
-## Enable `-parameters` compiler flag
-
-Most of our SQL Object tests rely on SQL method parameter names. However by default, `javac` does not compile these
-parameter names into `.class` files. Thus, in order for unit tests to pass, the compiler must be configured to output
-parameter names.
-
-### IntelliJ IDEA
-
-* File &rarr; Settings
-* Build, Execution, Deployment &rarr; Compiler &rarr; Java Compiler
-* Additional command-line parameters: `-parameters`
-* Click Apply, then OK.
-* Build &rarr; Rebuild Project
+# Development Setup
 
-### Eclipse IDE
-
-* Window &rarr; Preferences &rarr; Java &rarr; Compiler
-* Section *Classfile Generation*
-* Check box *Add variable attributes to generated class files (used by the debugger)*
-* Check box *Store information about method parameters (usable via reflection)*
-* Click Apply and close dialog
+Most modern IDEs configure themselves correctly by importing the Jdbi repository as an Apache Maven project. If necessary, install support for Apache Maven first.
 
 ## Code Style / Formatting Rules
 
 The project uses a set of code style and formatting rules. These are enforced by Checkstyle and PMD as part of the build cycle.
-We ask that you configure the project in your development environment to support these rules as closely as possible.
 
-The JDBI Eclipse formatting preferences are available for download [here](internal/policy/src/main/resources/ide/jdbi-eclipse-formatter.xml).
+We do not review or merge PRs that do not pass our pre-merge checks.
 
 ### IntelliJ IDEA
 
-Import the file with the 'Adapter for Eclipse Code Formatter'. Additional info is available [here](https://plugins.jetbrains.com/plugin/6546-adapter-for-eclipse-code-formatter).
+* [IntelliJ IDEA formatting rules](https://github.com/jdbi/jdbi/blob/master/ide-support/intellij/jdbi.xml).
```

Comment:
Rename the IDEA settings file to `jdbi-intellij.xml` for alignment with names of the Eclipse settings files?
You could then do away with the `intellij` / `eclipse` sub directories (but it's a matter of taste).

Reply:
the file name is used by the IDE.